pytest_flake8.py

```diff
@@ -188,11 +189,23 @@ def check_file(path, flake8ignore, maxlength, maxcomplexity,
     if statistics:
         args += ['--statistics']
     app = application.Application()
-    app.parse_preliminary_options_and_args(args)
-    app.make_config_finder()
-    app.find_plugins()
-    app.register_plugin_options()
-    app.parse_configuration_and_cli(args)
+    if not hasattr(app, 'parse_preliminary_options_and_args'):  # flake8 >= 3.8
+        prelim_opts, remaining_args = app.parse_preliminary_options(args)
+        config_finder = config.ConfigFileFinder(
+            app.program,
+            prelim_opts.append_config,
+            config_file=prelim_opts.config,
+            ignore_config_files=prelim_opts.isolated,
+        )
+        app.find_plugins(config_finder)
+        app.register_plugin_options()
+        app.parse_configuration_and_cli(config_finder, remaining_args)
+    else:
+        app.parse_preliminary_options_and_args(args)
+        app.make_config_finder()
+        app.find_plugins()
+        app.register_plugin_options()
+        app.parse_configuration_and_cli(args)
```

Comment:
this is still using private implementation details of flake8

Reply:
what's wrong with that? :thinking: 

It has been so since this plugin was created. This PR doesn't pretend to change the way the package works, but to add support for the last version.

Reply:
The goal of the patch is not to get rid of that, so hopefully that’s OK.

I think there should be a separate issue to address that.

Reply:
I mean, this breakage is a great example of exactly what's wrong with that 😆

Reply:
> Flake8 3.0.0 presently does not have a public, stable Python API.
> — https://flake8.pycqa.org/en/latest/user/python-api.html

Reply:
and just below that it shows how to use the legacy api

Reply:
I think your comments would be more useful in a new issue :sweat_smile: .

I will resolve this conversation, please, feel free to open the new issue giving your points of view. :bow: